Userland/Utilities/headless-browser.cpp

```diff
@@ -148,6 +149,16 @@ class HeadlessWebContentView final : public WebView::ViewImplementation {
             if (on_scroll_to_point)
                 on_scroll_to_point(position);
         };
+
+        on_request_worker_agent = []() {
+#if defined(AK_OS_SERENITY)
+            auto worker_client = MUST(Web::HTML::WebWorkerClient::try_create());
+#else
+            auto worker_client = MUST(launch_web_worker_process(MUST(get_paths_for_helper_process("WebWorker"sv))));
+#endif
```

Comment:
It's a bit unfortunate right now, because on Serenity we launch through SystemServer. We create the WebContent process with similar `#ifdef`s above (line 66).

A process launcher that handles SystemServer on serenity and forks on other platforms would be nice, but seems out of scope here.